### PR TITLE
nice

### PR DESCRIPTION
**Pull request by NikitaVysochin**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,37 +1,40 @@
 import logo from "./logo.svg";
 import "./App.scss";
 import React, { State, useEffect, useState } from "react";
+import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
 import axios from "axios";
 import Header from "./Header/Header";
 import MainInput from "./MainInput/MainInput";
 import Tasks from "./Tasks/Tasks";
+import Routing from "./RoutingRedact/Routing";
+
```
Comment (usatovaDasha) [inline src/App.js:10]: remove empty lines

File: src/App.js [hunk 2]
```diff
@@ -1,37 +1,40 @@
...
+        <Route path='/' element={<Tasks 
+          arr={arr} 
+          setArr={setArr} 
+          elem={elem}
+          setElem={setElem} 
+        />}/>
+        <Route path='/routing' element={<Routing 
+          elem={elem}
+          setElem={setElem}/>}/>
+      </Routes>
```
Comment (usatovaDasha) [inline src/App.js:36]: redirect no need?

File: src/RoutingRedact/Routing.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+const Routing = ({arr, setArr, elem, setElem}) => {
+  const [value, setValue] = useState(elem.elem);
+
+  const onBlur = async () => {
+    await axios
+      .patch("http://localhost:8000/updateTask", {
+        taskName: value,
+        isCheck: false,
+        _id: elem._id
```
Comment (usatovaDasha) [inline src/RoutingRedact/Routing.js]: destructurization elem

File: src/RoutingRedact/Routing.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      })
+      .then((res) => {
+        setArr(res.data.data);
+      });
+  };
+
+  return (
+    <div className='main-route'>
+      <Link to='/'><div className='button-Cancel'>Cancel</div></Link>
```
Comment (usatovaDasha) [inline src/RoutingRedact/Routing.js:24]: why need div?

File: src/RoutingRedact/Routing.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      <input value={value} onChange={(e) => setValue(e.target.value)}></input>
+      <Link to='/'><div className='button-return' onClick={onBlur}>Save</div></Link>
```
Comment (usatovaDasha) [inline src/RoutingRedact/Routing.js:26]: why need and link, and onClick?

File: src/RoutingRedact/Routing.scss [hunk 6]
```diff
@@ -0,0 +1,35 @@
+.main-route{
```
Comment (usatovaDasha) [inline src/RoutingRedact/Routing.scss]: space

File: src/RoutingRedact/Routing.scss [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    text-align: center;
+    line-height: 30px;
+    border-radius: 5px;
+    cursor: pointer;
+  }
+
+  input{
+    width: 60%;
+    text-align: center;
+  }
```
Comment (usatovaDasha) [inline src/RoutingRedact/Routing.scss:24]: add empty line

File: src/Tasks/Tasks.js [hunk 8]
```diff
@@ -39,7 +40,12 @@ const Tasks = ({ arr, setArr }) => {
   };
 
   const Redact = (elem, index) => {
-    setRed(index);
+    setElem({
+      index: index,
+      elem: elem.taskName,
```
Comment (usatovaDasha) [inline src/Tasks/Tasks.js]: destructurization

File: src/Tasks/Tasks.js [hunk 9]
```diff
@@ -39,7 +40,12 @@ const Tasks = ({ arr, setArr }) => {
...
+      _id: elem._id
+    });
+    //setRed(index);
```
Comment (usatovaDasha) [inline src/Tasks/Tasks.js]: remove all comments

File: src/index.js [hunk 10]
```diff
@@ -3,11 +3,12 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import {BrowserRouter as Router} from 'react-router-dom';
```
Comment (usatovaDasha) [inline src/index.js]: spaces

File: src/RoutingRedact/Routing.js (hunk 4 above)
Comment (NikitaVysochin) [reply src/RoutingRedact/Routing.js:24]: to discern input on white background)

File: src/RoutingRedact/Routing.js (hunk 5 above)
Comment (NikitaVysochin) [reply src/RoutingRedact/Routing.js:26]: onClick fulfill patch function. Link replace to main page

**Pull request merged**